File: app/src/main/java/com/fueled/reclaim/samples/MainActivity.java
package com.fueled.reclaim.samples;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.Menu;
import android.view.MenuItem;

import com.fueled.reclaim.ItemsViewAdapter;
import com.fueled.reclaim.samples.handler.ItemHandlerActivity;
import com.fueled.reclaim.samples.hearder.HearderFooterActivity;

import java.util.ArrayList;
import java.util.List;


public class MainActivity extends AppCompatActivity {

    private RecyclerView recyclerView;
    private ItemsViewAdapter adapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        recyclerView = (RecyclerView) findViewById(R.id.recyclerview);
    }

    @Override
    protected void onResume() {
        super.onResume();
        LinearLayoutManager manager = new LinearLayoutManager(this);
        manager.setOrientation(LinearLayoutManager.VERTICAL);
        recyclerView.setLayoutManager(manager);

        //set adapter
        adapter = new ItemsViewAdapter(this);
        adapter.replaceItems(createListData());

        recyclerView.setAdapter(adapter);
    }

    private List<PlanetItem> createListData() {
        ArrayList<PlanetItem> list = new ArrayList<>();
        list.add(new PlanetItem("Venus"));
        list.add(new PlanetItem("Mercury"));
        list.add(new PlanetItem("Earth"));
        list.add(new PlanetItem("Mars"));
        return list;
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.menu_main, menu); //Menu Resource, Menu
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case R.id.headerMenu:
                startActivity(new Intent(this, HearderFooterActivity.class));
                finish();
                return true;
            case R.id.handlerMenu:
                startActivity(new Intent(this, ItemHandlerActivity.class));
                finish();
                return true;
            default:
                return super.onOptionsItemSelected(item);
        }
    }
}

File: app/src/main/kotlin/com/fueled/reclaim/samples/MainActivity.kt
package com.fueled.reclaim.samples

import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.Observer
import androidx.recyclerview.widget.RecyclerView
import androidx.recyclerview.widget.StaggeredGridLayoutManager

import com.fueled.reclaim.ItemsViewAdapter
import com.fueled.reclaim.samples.items.HeaderAdapterItem
import com.fueled.reclaim.samples.items.NoteAdapterItem
import com.fueled.reclaim.samples.items.LoadingAdapterItem
import com.fueled.reclaim.samples.notes.Note
import com.fueled.reclaim.samples.notes.NotesStorage


class MainActivity : AppCompatActivity() {

    private lateinit var recyclerView: RecyclerView
    private val adapter = ItemsViewAdapter()

    private val cardColors = listOf(0xffe1f5fe, 0xfffffde7, 0xfffbe9e7, 0xffefebe9, 0xfff3e5f5)

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        recyclerView = findViewById(R.id.recycler_view)

        findViewById<View>(R.id.add_button).setOnClickListener {
            AddNoteDialog().show(supportFragmentManager, "ADD_NOTE_DIALOG")
        }

        val layoutManager = StaggeredGridLayoutManager(2, RecyclerView.VERTICAL)

        recyclerView.layoutManager = layoutManager

        //set adapter
        recyclerView.adapter = adapter

        subscribeToNoteUpdates()
    }

    private fun subscribeToNoteUpdates() {
        NotesStorage.getAllNotes()
            .observe(this, Observer {
                displayNotes(it)
            })
    }

    private fun displayNotes(notes: List<Note>) {
        val items = listOf(HeaderAdapterItem("Elliot Alderson")) + notes
            .mapIndexed { index, note ->
                NoteAdapterItem(note, cardColors[index % cardColors.size].toInt(), ::onDeleteNote)
            } + listOf(LoadingAdapterItem())

        adapter.replaceItems(
            items = items,
            useDiffCheck = true
        )
    }

    private fun onDeleteNote(note: Note) {
        NotesStorage.removeNote(note.id)
    }
}
